2.x feature/2023 01 19 vaahcms backend module development

wri-nitin-k001 opened the pull request:
<!-- PLEASE READ THE FOLLOWING INSTRUCTIONS -->

<!--
Make sure you provide all information and followed all the steps. Then only submit the Merge Request
-->

### List of all the tasks

Format of information to be provided

---

#### [Task #230 - VaahCms > Backend > Extends > Modules > Development](https://team.webreinvent.com/backend/collab/project/vaahcms/tasks/230)
**Time invested:** hh:mm format | **Billable**: 22:37 | **Non-Billable**: hh:mm

**Client's comment or requirement**

**Proof of your testing, video, or image link?**
Module section development - https://www.dropbox.com/s/jd4mg1semihavnh/chrome_VnyQtsJrHi.mp4?dl=0

### Self Review Checklist

#### Naming Conventions
> For a detailed information click [here](https://docs.vaah.dev/guide/code.html#naming-conventions)

- [x] Are your Controller names `meaningful`, `singular` and have `Controller` suffix?
- [x] Are your Model names `singular` and named after the table they are modelling?
- [x] Are your Table names `snake_case` and `plural`?
- [x] Are your Table Column names `snake_case` without model name?
- [ ] Are your Seeder names `descriptive` about which table they are seeding?
- [x] Are your method/function names `short`, `descriptive` and `camelCased`?
- [x] Are your variable names `short`, `descriptive` and `snake_cased`?

#### Constants and Configs
> For a detailed information click [here](https://docs.vaah.dev/guide/code.html#naming-conventions)

- [ ] API keys, Secret Keys, URL's etc., are stored as env variables.
- [ ] I am accessing env variables via config.
- [ ] I have used config and language files, constants instead of text in the code .
- [ ] I have not hardcoded any value.

#### Fat methods  and CRUD
> For a detailed information click [here](https://docs.vaah.dev/guide/code.html#laravel-community-guidelines-for-good-coding-practices)

- [x] All requests are being validated
- [x] I am not using raw SQL queries. I am using `Eloquent` queries.
- [x] DB related logic are in `Eloquent models` or in `Repository`/`Helper` classes.
- [x] I have not included business logic in Controllers. They are in `Repository`/`Helper` classes.
- [ ] My methods adhere to the `Single Responsibility Principle`.
- [x] My code doesn't contain extensive if-else nesting. I am checking negative conditions first.
- [x] I have removed all unwanted commented code.
- [ ] I have done my tech debt analysis and any unavoidable code leading to tech debt is marked with `@todo` .

#### Other practices

- [x] All my migrations have `down` method that reverses operations performed by `up` method.
- [ ] Any new column added/altered in table has been also updated in the `fillable` property of its model.
- [ ] For seeders that are not populating sample data, it is not truncating table before seeding again.
- [ ] For seeders that are not populating sample data, the data is being populated from a separate external file.
- [ ] There are no business logic or queries in routes.
- [x] I have not put queries in Blade templates.
- [x] I have not put JS and CSS in Blade templates and have not put any HTML in PHP classes.
- [x] I have removed all debug code.

#### Nuxt coding practices

- [ ] My Vue Components names are `short`, describes what it's intended use is and is in `PascalCase`
- [ ] All the components exclusive to the page are in the page's `components` folder
- [ ] I have preferred using Composition API.
- [ ] I have preferred using `<script setup>` tag instead of the `setup()`.
- [ ] I have used asyncData or fetch hook (avoid mounted or created hooks) to fetch any initial data that will make the component load later than page.
- [ ] I have marked any component that does not have reactive elements as `functional`.
- [x] I have used `key` for any `v-for` used.
- [x] I have not used `v-for` and `v-if` in the same element.
- [x] I have declared `props` with `camelCase` but used in templates with `kebab-case`.
- [x] I have validated my `props` with good definition, and have set there default values
- [ ] I have added a `Readme.md` file for each component I have created.
- [x] All my page specific methods and variables are in the page's store.

---

### Feature Merge Request Checklist

- [x] have you rebased your `feature` with `develop`?
- [ ] have run `npm run prod` after `rebase` with `develop` branch in all your `Modules` & `Themes`? Notes, you have to reactivate modules and themes to publish assets.
- [ ] have you updated the `patch` (`x.x.<update-this>`) version `config.php` and `composer.json` of module & theme?
- [x] have you read all the `comments` & `notes` in `wireframe` and verified that you have taken care of that?
- [x] verify that the `UI` must match with `design` & `wireframe` if available?
- [x] have you applied `data-wdio` tag to `links`, `forms` & all `form inputs` like `input`, `select`?
- [x] did you verify the latest commit of `develop` exists in your `feature` branch after rebase?
- [x] have you tested the `feature` as per the client requirement?
- [x] did you follow the proper naming conventions in your code?
- [ ] have you commented on your code with proper explanation?

we-vikram-wri231 committed:
Added: log CRUD file

we-vikram-wri231 committed:
Updated: all action in view page

we-vikram-wri231 committed:
Updated: delete action in list page

Commit:
Advance > Batches section converted to vue3

Batches crud created.
ui configured according to old. 
new routes, controller and models created.
old model renamed  and moved to model folder from entities.
old controller deleted.

wri-abinash-s001 committed:
Update: Failed-job crud and modified

wri-abinash-s001 committed:
Update: Failed-job crud and modified

wri-abinash-s001 committed:
Update: Failed-job crud and modified

wri-abinash-s001 committed:
Update: Failed-job crud and modified

Commit:
advanced route file changed. batches imports updated

wri-abinash-s001 committed:
Update: Failed-Job crud from vue2 to vue3 complete

wri-abinash-s001 committed:
Update: Failed-Job crud from vue2 to vue3 complete

wri-abinash-s001 committed:
Update: Failed-Job crud from vue2 to vue3 complete

wri-abinash-s001 committed:
Update: Failed-Job crud from vue2 to vue3 complete

Commit:
popup, progressbar added, routes changes made,soft delete code removed from batch model

Commit:
popup data format changed to json. fillable columns added to BatchBase Model. filters added

Commit:
popup content stringified. column made non sortable.

Commit:
Theme Backend crud Ui setup done

Theme Backend crud Ui setup done. old model moved to models folder and renamed with Base as suffix and imports refactored for the old model.
controller code updated to display and create required data.

Commit:
Theme Backend crud Ui setup done

Theme Backend crud Ui setup done. old model moved to models folder and renamed with Base as suffix and imports refactored for the old model.
controller code updated to display and create required data.

we-vikram-wri231 committed:
Merge remote-tracking branch 'origin/2.x-feature/Failed-Job-crud' into 2.x-feature/Failed-Job-crud

we-vikram-wri231 committed:
Updated: failed jobs file in advanced folder

we-vikram-wri231 committed:
Updated: failed jobs file in advanced folder

Commit:
List UI integrated for themes section. Activate, Deactivate and Make Default Actions integrated.

Commit:
List UI integrated for themes section. Activate, Deactivate and Make Default Actions integrated. 

List UI integrated for themes section. Activate, Deactivate and Make Default Actions integrated. Extra Import Removed.

Commit:
List UI integrated for themes section. Activate, Deactivate and Make Default Actions integrated. 

List UI integrated for themes section. Activate, Deactivate and Make Default Actions integrated. Extra Import Removed.

Commit:
List UI integrated for themes section. Activate, Deactivate and Make Default Actions integrated.

Commit:
List UI Page integrated for themes section.

List UI integrated for themes section. Activate, Deactivate and Make Default Actions integrated. Extra Import Removed.

wri-abinash-s001 committed:
Update: user-settings modification list fetched

wri-abinash-s001 committed:
Update: user-settings modification list fetched

wri-abinash-s001 committed:
Update: user-settings modification list fetched

wri-abinash-s001 committed:
Update : field section fetch complete

wri-abinash-s001 committed:
Update : field section fetch complete

wri-abinash-s001 committed:
Update : field section fetch complete

Commit:
Theme install UI and Actions Integrated

Theme install UI integrated.
Sample Data Import Action integrate.
SIngle Delete Action integrated.
Install and CheckUpdates Actions integrated.

Commit:
Theme install UI and Actions Integrated

Theme install UI integrated.
Sample Data Import Action integrate.
SIngle Delete Action integrated.
Install and CheckUpdates Actions integrated.

wri-abinash-s001 committed:
Update: onload error fixed

wri-abinash-s001 committed:
Update: onload error fixed

wri-abinash-s001 committed:
Update: onload error fixed

wri-abinash-s001 committed:
Update: custom field and draggable complete

wri-abinash-s001 committed:
Update: custom field and draggable complete

wri-abinash-s001 committed:
Update: custom field and draggable complete

Commit:
themes Vue 3 Changes

filters integrated
delete confimation added
download theme issue resolved

Commit:
themes Vue 3 Changes

filters integrated
delete confimation added
download theme issue resolved

Commit:
Install Button switch completed.
Paginator added

Commit:
Install Button switch completed.
Paginator added

wri-abinash-s001 committed:
Update: user-setting modification complete

wri-abinash-s001 committed:
Update: user-setting modification complete

wri-abinash-s001 committed:
Update: user-setting modification complete

wri-abinash-s001 committed:
Update: general setting UI and controller setup complete

wri-abinash-s001 committed:
Update: general setting UI and controller setup complete

Commit:
Modules section created

Module List Ui integrated.
Module Activate, Sample data import action integrated.

Commit:
Modules section created

Module List Ui integrated.
Module Activate, Sample data import action integrated.

wri-abinash-s001 committed:
Update: general-setting complete need to verify with old

wri-abinash-s001 committed:
Update: general-setting complete need to verify with old

Commit:
Module Section functionality completed

FIlters integrated
Activate functionality integrated
Sample data import functionality integrated
update module functionality integrated
Delete functionality integrated
module Vew Secetion integrated
install module section integrated
install module functionality integrated
check update functionality integrated
reload list functionality integrated

Commit:
Module Section functionality completed

FIlters integrated
Activate functionality integrated
Sample data import functionality integrated
update module functionality integrated
Delete functionality integrated
module Vew Secetion integrated
install module section integrated
install module functionality integrated
check update functionality integrated
reload list functionality integrated

wri-tanmoy-d001 committed:
Added: taxonomy related crud

wri-tanmoy-d001 committed:
Added: taxonomy related crud

wri-tanmoy-d001 committed:
Added: taxonomy related crud

wri-tanmoy-d001 committed:
Added: taxonomy related crud

we-vikram-wri231 committed:
Fixed: user setting

wri-tanmoy-d001 committed:
Refactor: vue and laravel files

wri-tanmoy-d001 committed:
Refactor: vue and laravel files

wri-tanmoy-d001 committed:
Refactor: vue and laravel files

wri-tanmoy-d001 committed:
Refactor: vue and laravel files

we-vikram-wri231 committed:
Updated: view condition and tag

we-vikram-wri231 committed:
Updated: tag classes

we-vikram-wri231 committed:
Updated: tag classes

wri-abinash-s001 committed:
fixed: variable name

wri-abinash-s001 committed:
fixed: variable name

wri-abinash-s001 committed:
fixed: variable name

wri-abinash-s001 committed:
fixed: variable name

we-vikram-wri231 committed:
Updated: document title

we-vikram-wri231 committed:
Updated: document title

we-vikram-wri231 committed:
Updated: document title

we-vikram-wri231 committed:
Updated: document title

wri-abinash-s001 committed:
Update: meta tag in general functionality

wri-abinash-s001 committed:
Update: meta tag in general functionality

we-vikram-wri231 committed:
Merge remote-tracking branch 'origin/2.x-feature/User-devlopment' into 2.x-feature/User-devlopment

Commit:
Module UI updated

Commit:
Module UI updated

wri-tanmoy-d001 committed:
Updated: UI and fixed some console error

wri-tanmoy-d001 committed:
Updated: UI and fixed some console error

wri-tanmoy-d001 committed:
Updated: UI and fixed some console error

wri-tanmoy-d001 committed:
Updated: UI and fixed some console error

wri-tanmoy-d001 committed:
Merge remote-tracking branch 'origin/2.x-feature/taxonomy-development' into 2.x-feature/taxonomy-development

Commit:
data test ids updated

Commit:
data test ids updated

wri-tanmoy-d001 committed:
Added: manage taxonomy type modal

wri-tanmoy-d001 committed:
Added: manage taxonomy type modal

wri-abinash-s001 committed:
Fixed: draggable div design change and variable name

wri-abinash-s001 committed:
Fixed: draggable div design change and variable name

Commit:
Themes section updated

themes ui updated
permissions added to themes

Commit:
Themes section updated

themes ui updated
permissions added to themes

wri-tanmoy-d001 committed:
Updated: manage taxonomy type modal

wri-tanmoy-d001 committed:
Updated: manage taxonomy type modal

Commit:
Modules Extra Code Removed

Commit:
Modules Extra Code Removed

Commit:
Themes Extra Code Removed

Commit:
Themes Extra Code Removed

wri-tanmoy-d001 committed:
Updated: form.vue and check permission on create taxonomy

wri-tanmoy-d001 committed:
Updated: form.vue and check permission on create taxonomy

wri-tanmoy-d001 committed:
Updated: check permission on create button

wri-tanmoy-d001 committed:
Updated: check permission on create button

wri-tanmoy-d001 committed:
Fixed: is active select button

wri-tanmoy-d001 committed:
Fixed: is active select button

wri-tanmoy-d001 committed:
Added: sync method

wri-tanmoy-d001 committed:
Added: sync method

wri-abinash-s001 committed:
Update: site setting autocomplete inprogress

wri-abinash-s001 committed:
Update: site setting autocomplete inprogress

wri-tanmoy-d001 committed:
Added: create taxonomy type method

wri-tanmoy-d001 committed:
Added: create taxonomy type method

Commit:
Media Crud Setup

created crud
integrated create ui with the new Ui
integrated list component

Commit:
Media Crud Setup

created crud
integrated create ui with the new Ui
integrated list component

wri-tanmoy-d001 committed:
Added: laravel methods

wri-tanmoy-d001 committed:
Added: laravel methods

wri-tanmoy-d001 committed:
Removed: unwanted imports

wri-tanmoy-d001 committed:
Removed: unwanted imports

wri-tanmoy-d001 committed:
Added: create taxonomy type method

wri-tanmoy-d001 committed:
Added: create taxonomy type method

wri-tanmoy-d001 committed:
Fixed: taxonomy layout

wri-tanmoy-d001 committed:
Fixed: taxonomy layout

wri-abinash-s001 committed:
Update: General setting complete

wri-abinash-s001 committed:
Update: General setting complete

wri-tanmoy-d001 committed:
Fixed: filter's sidebar z-index

wri-tanmoy-d001 committed:
Fixed: filter's sidebar z-index

wri-abinash-s001 committed:
Merge remote-tracking branch 'origin/2.x-develop' into 2.x-develop

wri-tanmoy-d001 committed:
Updated: manage taxonomy type modal (wip)

wri-tanmoy-d001 committed:
Updated: manage taxonomy type modal (wip)

wri-abinash-s001 committed:
Update: notification code setup and UI is complete

Commit:
Media Section

filters functionality updated
create form functionality updated
delete funcitonality updated

Commit:
Media Section

filters functionality updated
create form functionality updated
delete funcitonality updated

themodernpk commented:
@we-vikram-wri231 @wri-nitin-k001 review the feedback https://drive.google.com/file/d/1uxthKVc8lqMoexxSE_BDFW3XevtjCbQs/view

Commit:
Changes Made to Modules List and Install page section.

Image Strech style removed from Install Page item's image.
assets publish Button added to List item.

Commit:
Changes Made to Modules List and Install page section.

Image Strech style removed from Install Page item's image.
assets publish Button added to List item.

Commit:
VaahSetup::publishAssets method caling replaced with VaahArtisan::publishAssets

Commit:
VaahSetup::publishAssets method caling replaced with VaahArtisan::publishAssets

wri-nitin-k001 commented:
> @we-vikram-wri231 @wri-nitin-k001 review the feedback https://drive.google.com/file/d/1uxthKVc8lqMoexxSE_BDFW3XevtjCbQs/view
ok sir. I'll do it.

Commit:
Filters and updated media functionality updated

trashed filter updated
date range functionality updated.
update media functionality updated

Commit:
Filters and updated media functionality updated

trashed filter updated
date range functionality updated.
update media functionality updated

Commit:
Filter lable changed

Commit:
Filter lable changed

wri-abinash-s001 committed:
Update: setting-notification complete need testing

Commit:
Media FIle upload  component changed.

Commit:
Media FIle upload  component changed.

Commit:
changes made to file upload  component
data-testid's  updated

Commit:
changes made to file upload  component
data-testid's  updated

Commit:
published assets functionality updated.

Commit:
published assets functionality updated.

wri-abinash-s001 committed:
Update: setting notification complete

Commit:
file upload url attribute error  fixed.

Commit:
file upload url attribute error  fixed.

Commit:
extra debug code removed.

Commit:
extra debug code removed.

Commit:
label changed. double click to change select images issue fixed.

Commit:
label changed. double click to change select images issue fixed.

Commit:
FIle Upload  label changed

Commit:
FIle Upload  label changed

Commit:
extra controller class removed.

Commit:
extra controller class removed.

wri-abinash-s001 committed:
Update: check and removed console error

wri-abinash-s001 committed:
Update: check and removed console error

Commit:
changes made to filters

Commit:
changes made to filters

wri-abinash-s001 committed:
Update: setting update  conversion almost completed

Commit:
changes  made to date filter. Changes made to popup window for Details and Failed ids

Commit:
changes  made to date filter. Changes made to popup window for Details and Failed ids

Commit:
radio Button corrected

wri-tanmoy-d001 committed:
Removed: commented codes and unnecessary files

wri-tanmoy-d001 committed:
Removed: commented codes and unnecessary files

wri-tanmoy-d001 committed:
Updated: copy slug icon

wri-tanmoy-d001 committed:
Updated: copy slug icon

wri-tanmoy-d001 committed:
Updated: taxonomy controller's assets method

wri-tanmoy-d001 committed:
Updated: taxonomy controller's assets method

wri-abinash-s001 committed:
Update: setting update dynamically using terminal complete

Commit:
change made to date filter backend code. Delete issue fixed.

Commit:
right margin added to  view icons

Commit:
data test ids updated

Commit:
duplicate code removed from model

Commit:
filters changed, 
pagination issue fixed. 
File upload ui changed.

Commit:
filters changed, 
pagination issue fixed. 
File upload ui changed.

wri-tanmoy-d001 committed:
Fixed: Tree Select input

wri-tanmoy-d001 committed:
Fixed: Tree Select input

wri-tanmoy-d001 committed:
Fixed: parent_id tree select on create taxonomy form

wri-tanmoy-d001 committed:
Fixed: parent_id tree select on create taxonomy form

wri-abinash-s001 committed:
Update: sticky upgrade notification complete

wri-tanmoy-d001 committed:
Added: parent country input field

wri-tanmoy-d001 committed:
Added: parent country input field

we-vikram-wri231 committed:
Merge remote-tracking branch 'origin/2.x-feature/Failed-Job-crud' into 2.x-feature/Failed-Job-crud

we-vikram-wri231 committed:
Fixed: conflict issue

wri-tanmoy-d001 committed:
Updated: city's parent id drop down

wri-tanmoy-d001 committed:
Updated: city's parent id drop down

wri-abinash-s001 committed:
Fixed: exception modal format fixed

wri-abinash-s001 committed:
Merge remote-tracking branch 'origin/2.x-feature/Failed-Job-crud' into 2.x-feature/Failed-Job-crud

we-vikram-wri231 committed:
Merge pull request #93 from webreinvent/2.x-feature/Failed-Job-crud

2.x feature -> 2.x-develop | failed-job development

we-vikram-wri231 committed:
Fixed: log conflict

we-vikram-wri231 committed:
Merge branch '2.x-feature/log-crud' into 2.x-develop

we-vikram-wri231 committed:
Fixed: conflict

we-vikram-wri231 committed:
Removed: batches route

we-vikram-wri231 committed:
Merge branch '2.x-feature/vaahcms-advanced-batch-development' into 2.x-develop

we-vikram-wri231 committed:
Merge remote-tracking branch 'origin/2.x-feature/taxonomy-development' into 2.x-feature/taxonomy-development

we-vikram-wri231 committed:
Merge branch '2.x-feature/taxonomy-development' into 2.x-develop

we-vikram-wri231 committed:
Merge remote-tracking branch 'origin/2.x-feature/2023-01-24-VaahCms-backend-media-development' into 2.x-feature/2023-01-24-VaahCms-backend-media-development

we-vikram-wri231 committed:
Fixed: conflict

we-vikram-wri231 committed:
Merge branch '2.x-feature/2023-01-24-VaahCms-backend-media-development' into 2.x-develop

we-vikram-wri231 committed:
Merge remote-tracking branch 'origin/2.x-feature/General-devlopment' into 2.x-feature/General-devlopment

we-vikram-wri231 committed:
Merge branch '2.x-feature/General-devlopment' into 2.x-develop

we-vikram-wri231 committed:
Merge remote-tracking branch 'origin/2.x-feature/User-devlopment' into 2.x-feature/User-devlopment

we-vikram-wri231 committed:
Fixed: conflict error

we-vikram-wri231 committed:
Merge branch '2.x-feature/User-devlopment' into 2.x-develop

we-vikram-wri231 committed:
Merge remote-tracking branch 'origin/2.x-feature/Notification-devlopment' into 2.x-feature/Notification-devlopment

we-vikram-wri231 committed:
Merge branch '2.x-feature/Notification-devlopment' into 2.x-develop

we-vikram-wri231 committed:
Merge remote-tracking branch 'origin/2.x-feature/2023-01-13-vaacms-backend-theme-development' into 2.x-feature/2023-01-13-vaacms-backend-theme-development

we-vikram-wri231 committed:
Merge branch '2.x-feature/2023-01-13-vaacms-backend-theme-development' into 2.x-develop

we-vikram-wri231 committed:
Fixed: conflict

we-vikram-wri231 committed:
Merge branch '2.x-feature/2023-01-13-vaacms-backend-theme-development' into 2.x-develop

we-vikram-wri231 committed:
Merge remote-tracking branch 'origin/2.x-feature/2023-01-19-vaahcms-backend-module-development' into 2.x-feature/2023-01-19-vaahcms-backend-module-development

we-vikram-wri231 committed:
Fixed: extend module conflicts
